deep_qa/layers/wrappers/add_encoder_mask.py

```diff
@@ -0,0 +1,55 @@
+from keras import backend as K
+from overrides import overrides
+
+from ..masked_layer import MaskedLayer
+
+
+class AddEncoderMask(MaskedLayer):
+    """
+    This ``Layer`` handles masking for ``TimeDistributed`` encoders, like LSTMs.  Our
+    :class:`~.encoder_wrapper.EncoderWrapper` class does the correct masking computation, but it
+    inherits from ``TimeDistributed``, which does not work with unknown dimensions at run-time.  If
+    you want to wrap an encoder using :class:`~..backend.CollapseToBatch` and
+    :class:`~..backend.ExpandFromBatch`, you need a way to get the mask back into the right form
+    after running your encoder.  This is an issue because Keras' encoders don't return masks when
+    they output single vectors.
+
+    For example, say you have a list of sentences, like [[5, 2, 1, 0], [2, 3, 1, 1], [0, 0, 0, 0]]
+    (using word indices instead of embeddings for simplicity), which has been padded to be three
+    sentences, even though only two of them are actually used.  After passing it though an encoder,
+    you'll have something like [[vector], [vector], [vector]], and you want a mask that looks like
+    [1, 1, 0].  Keras' LSTMs and such won't give this to you.  This method adds it back.
+
+    Inputs:
+        - A tensor with shape ``(batch_size, ..., encoding_dim)`` that is the output of some encoder.
+          The mask for this tensor must be ``None``.
+        - A tensor with shape ``(batch_size, ..., num_words, embedding_dim)`` that was the `input`
+          to that encoder.  The mask for this tensor must have shape ``(batch_size, ...,
+          num_words)``.
+
+    Output:
+        - The first input tensor, with a mask computed from the second input tensor.  The
+          computation is just ``K.any()`` on the last dimension.
+    """
+    def __init__(self, **kwargs):
+        # The base layer class overwrites self.supports_masking,
+        # so we add it afterwards.
```

Comment:
Thanks for catching that.  It's a copy-paste comment that was still in `OutputMask` because I forgot to remove it when I made it inherit from `MaskedLayer`.  Removed here and in `OutputMask`.